src/Bridge/Doctrine/Orm/Filter/AbstractFilter.php

```diff
@@ -308,13 +308,14 @@ protected function addJoinsForNestedProperty(string $property, string $rootAlias
     /**
      * Gets the Doctrine Type of a given property/resourceClass.
      *
-     * @return Type|string|null
+     * @return string|null
      */
     protected function getDoctrineFieldType(string $property, string $resourceClass)
     {
         $propertyParts = $this->splitPropertyParts($property, $resourceClass);
         $metadata = $this->getNestedMetadata($resourceClass, $propertyParts['associations']);
+        $type = $metadata->getTypeOfField($propertyParts['field']);
 
-        return $metadata->getTypeOfField($propertyParts['field']);
+        return null !== $type ? (string) $type : null;
```

Comment:
I wonder if this is to be consider as a BC break though...

Reply:
We're changing the return type so I guess it is, just wondering if we use `Type` at all, if not I would say that it's no big deal.

Reply:
We don't use Type but could be used by some custom filter maybe... In my own filters I always add to cast it so PhpStan is not compaining, so I too think it is not a big deal. Fixing this potential BC break will be easy enough also but will "cost" a release so I wanted to draw your attention on it.

Reply:
I think we should keep BC here. The calling code can do the cast.

Reply:
Also important to note: https://github.com/doctrine/dbal/blob/1b1adc2a8d4ab8dcfe5c328ccd3864a0fd36f6a7/lib/Doctrine/DBAL/Types/Type.php#L252

```php
    /**
     * @deprecated Relying on string representation is discouraged and will be removed in DBAL 3.0.
```

Reply:
I'll revert then

Reply:
Thanks @teohhanhui !

Reply:
ok it has been reverted.